homework-3/app.js

```diff
@@ -0,0 +1,106 @@
+const express = require("express");
+const bodyParser = require("body-parser");
+const uuid = require("uuid");
+
+const app = express();
+
+const fs = require("fs");
+const { json } = require("express");
+
+const port = 5000;
+
+const db = "./homework-3/db.json";
+
+app.use(bodyParser.urlencoded({ extended: false }));
+
+app.use(bodyParser.json());
+
+app.get("/products", (req, res) => {
+  fs.readFile(db, "utf8", (err, data) => {
+    if (err) throw err;
+    else {
+      let products = JSON.parse(data);
+      res.send(products);
+    }
+  });
+});
+
+app.post("/products", (req, res) => {
+  const { product } = req.body;
+  const parseProduct = JSON.stringify(product);
+  let productsPost;
+  fs.readFile(db, json, (err, data) => {
+    if (err) throw err;
+    else {
+      let products = JSON.parse(data);
+      products.push({ ...product, id: uuid.v4() });
+      productsPost = JSON.stringify(products);
+    }
+    fs.writeFile(db, productsPost, (err) => {
+      if (err) throw err;
+      res.send(parseProduct);
+    });
+  });
+});
+
+app.put("/products/:id", (req, res) => {
+  const { id } = req.params;
+  const { product } = req.body;
+  let productsPut;
+  fs.readFile(db, "utf8", (err, data) => {
+    if (err) throw err;
+    else {
+      let products = JSON.parse(data);
+      products = products.map((productFromDb) =>
+        id === productFromDb.id ? { ...product, id: uuid.v4() } : productFromDb
+      );
+      productsPut = JSON.stringify(products);
+    }
+    fs.writeFile(db, productsPut, (err) => {
+      if (err) throw err;
+      res.send(productsPut);
+    });
+  });
+});
+
+app.get("/products/:id", (req, res) => {
+  const { id } = req.params;
+  fs.readFile(db, "utf8", (err, data) => {
+    if (err) throw err;
+    else {
+      let products = JSON.parse(data);
+      products = products.filter((product) => product.id === id);
```

Comment:
It's not correct to use filter here. Filter returns you array of items that match your condition, which means it will check EVERY element in your array. What if I need to get user with id 1 and there is million users, it will take eternity